tests/database/test_header_db.py

```diff
@@ -242,3 +242,45 @@ def test_headerdb_canonical_header_retrieval_by_number(headerdb, genesis_header)
     for header in headers:
         actual = headerdb.get_canonical_block_header_by_number(header.block_number)
         assert_headers_eq(actual, header)
+
+
+def test_headerdb_header_exists(headerdb, genesis_header):
+    assert headerdb.header_exists(genesis_header.hash) is False
+    headerdb.persist_header(genesis_header)
+    assert headerdb.header_exists(genesis_header.hash) is True
+
+    chain_a = mk_header_chain(genesis_header, 3)
+    chain_b = mk_header_chain(genesis_header, 5)
+
+    assert not any(headerdb.header_exists(h.hash) for h in chain_a)
+    assert not any(headerdb.header_exists(h.hash) for h in chain_b)
+
+    for idx, header in enumerate(chain_a):
+        assert all(headerdb.header_exists(h.hash) for h in chain_a[:idx])
+        assert not any(headerdb.header_exists(h.hash) for h in chain_a[idx:])
+
+        # sanity pre-check
+        assert not headerdb.header_exists(header.hash)
+        headerdb.persist_header(header)
+        assert headerdb.header_exists(header.hash)
+
+    # `chain_a` should now all exist
+    assert all(headerdb.header_exists(h.hash) for h in chain_a)
+    # `chain_b` should not be in the database.
+    assert not any(headerdb.header_exists(h.hash) for h in chain_b)
+
+    for idx, header in enumerate(chain_b):
+        # `chain_a` should remain accessible
+        assert all(headerdb.header_exists(h.hash) for h in chain_a)
+
+        assert all(headerdb.header_exists(h.hash) for h in chain_b[:idx])
+        assert not any(headerdb.header_exists(h.hash) for h in chain_b[idx:])
+
+        # sanity pre-check
+        assert not headerdb.header_exists(header.hash)
+        headerdb.persist_header(header)
+        assert headerdb.header_exists(header.hash)
```

Comment:
Same with these two asserts.

Reply:
I'm wishing to leave them in for the same reason as the previous section.  Veto if you feel strongly.